### PR TITLE
Add action to set total pulses on pulse_meter

### PR DESCRIPTION
**Pull request by jesserockz**
# What does this implement/fix? 

Adds an action to set the total count of pulses 

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (this will require users to update their yaml configuration files to keep working)

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#1147
  
# Test Environment

- [x] ESP32
- [ ] ESP8266
- [ ] Windows
- [ ] Mac OS
- [ ] Linux

## Example entry for `config.yaml`:

```yaml
# Example config.yaml
on_...:
  - pulse_meter.set_total_pulses:
      id: pulse_meter_sensor
      value: 1000
```

# Explain your changes

Describe your changes here to communicate to the maintainers **why we should accept this pull request**.
Very important to fill if no issue linked

This action allows setting the total pulse count to match what your actual electricity or water meter might already have on it.

The idea behind this would be to add a [user-defined API service](https://esphome.io/components/api.html#user-defined-services) to receive the value to be set so it is not hardcoded in firmware

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).
  
If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Comment by m1ch** [comment]
@jesserockz I like the idea in general. If you have a value set in your yaml-file, it is hard-coded in the firmware. Each time the ESP reboots, it will be reset to this value. I did not see that you store the value in the eeprom. 

For this purpose you can also use already globals and update on any ON_-action. With globals you can decide to store them in the eprom, what might shorten the life of your device, or set them via MQTT. 

I have home-assitant publishing my last latest values on MQTT and I will update my globals on the on_boot action.

File: tests/test1.yaml [hunk 1]
```diff
@@ -658,9 +658,14 @@ sensor:
...
   - platform: pulse_meter
     name: 'Pulse Meter'
+    id: pulse_meter_sensor
     pin: GPIO12
     internal_filter: 100ms
     timeout: 2 min
+    on_value:
+      - pulse_meter.set_total_pulses:
+          id: pulse_meter_sensor
+          value: 12345
```
Comment (m1ch) [inline tests/test1.yaml:668]: I would recommend removing the option to set a hard coded value in the firmware. 
Just add the set_value_action.

Comment (jesserockz) [reply tests/test1.yaml:668]: This is just a test file to make sure the config passes.

**Comment by jesserockz** [comment]
> @jesserockz I like the idea, I misunderstood at first reading. Maybe add some more information in the description

I updated the description with: 
> The idea behind this would be to add a [user-defined API service](https://esphome.io/components/api.html#user-defined-services) to receive the value to be set so it is not hardcoded in firmware

**Review by stevebaxter — APPROVED** [review]
This will be super-useful!

**Pull request merged**